include/nbl/video/IDescriptorPool.h

```diff
@@ -107,14 +107,7 @@ class IDescriptorPool : public core::IReferenceCounted, public IBackendObject
 
         virtual ~IDescriptorPool()
         {
-            if (allowsFreeing())
-            {
-                assert(m_descriptorSetAllocator.get_allocated_size() == 0);
-#ifdef _NBL_DEBUG
-                for (uint32_t i = 0u; i < m_creationParameters.maxSets; ++i)
-                    assert(m_allocatedDescriptorSets[i] == nullptr);
-#endif
```

Comment:
don't get rid of this check, very useful

Reply:
I agree, but just to make this check pass we would have to do something like `std::fill_n` the `nullptr` in implicit `reset`. Because I don't think there is any other place where we can nullify the `m_allocatedDescriptorSets` pointers. Note that we do have `m_allocatedDescriptorSets[i] = nullptr` in `deleteSetStorage` but the call from `~IGPUDescriptorSet` won't nullify the `this` ptr.

Reply:
but doesn't `~IGPUDescriptorSet` call `deleteSetStorage` and that has the `m_allocatedDescriptorSets[getOffset] = nullptr` ?

Reply:
`reset()` can iterate the allocated DS-es and nullify them

Reply:
As discussed on Discord, this can be solved by letting the `deleteSetStorage` take an index into `m_allocatedDescriptorSets` rather than a `IGPUDescriptorSet*`. Resolved.